apps/ui/src/components/pages/application/ApplicationViewerHeader.tsx

```diff
@@ -80,6 +86,35 @@ const ApplicationViewerHeader = ({ id, appState, currentSection, isEditMode }: A
 		}
 	};
 
+	// Generate download url and then remove the link after downloading
+	const onPDFDownload = async () => {
+		const response = await getDownload();
+
+		const { data: responseData } = response;
+
+		// If there is no response data OR the file name does not exist, fail the download procedure
+		if (!responseData || responseData.filename === null) {
+			return;
+		}
+
+		const bufferArray = new Uint8Array(responseData.content.data).buffer;
+
+		const blob = new Blob([bufferArray], {
+			type: 'pdf',
+		});
+
+		const url = URL.createObjectURL(blob);
+		const a = document.createElement('a');
+		a.href = url;
+
+		a.download = responseData.filename;
+		document.body.appendChild(a);
+		a.click();
+
+		document.body.removeChild(a);
+		URL.revokeObjectURL(url);
```

Comment:
This is helpful, appreciate you adding this. Probably overkill and unlikely to ever make a difference, but still good housekeeping.